admiral/pkg/apis/admiral/routes/handlers.go

```diff
@@ -25,4 +38,106 @@ func (opts *RouteOpts) ReturnSuccessGET(w http.ResponseWriter, r *http.Request)
 
 func (opts *RouteOpts) GetClusters(w http.ResponseWriter, r *http.Request) {
 
+	clusterList := []string{}
+
+	// loop through secret controller's c.cs.remoteClusters to access all clusters admiral is watching
+	for clusterID := range opts.RemoteRegistry.SecretController.Cs.RemoteClusters {
+		clusterList = append(clusterList, clusterID)
+	}
+
+	out, err := json.Marshal(clusterList)
+	if err != nil {
+		log.Printf("Failed to marshall response for GetClusters call")
+		http.Error(w, "Failed to marshall response", http.StatusInternalServerError)
+	} else {
+		if len(clusterList) == 0 {
+			message := "No cluster is monitored by admiral"
+			log.Printf(message)
+			w.WriteHeader(200)
+			out, _ = json.Marshal(message)
+			w.Write(out)
+		} else {
+			w.Header().Set("Content-Type", "application/json")
+			w.WriteHeader(200)
+			w.Write(out)
+		}
+	}
+}
+
+func (opts *RouteOpts) GetServiceEntriesByCluster(w http.ResponseWriter, r *http.Request) {
+	defer r.Body.Close()
+
+	params := mux.Vars(r)
+	clusterName := strings.Trim(params["clustername"], " ")
+
+	var response []v1alpha3.ServiceEntry
+
+	if clusterName != "" {
+
+		serviceEntriesByCluster, err := clusters.GetServiceEntriesByCluster(clusterName, opts.RemoteRegistry)
+
+		if err != nil {
+			log.Printf(err.Error())
```

Comment:
Print some request context, would be hard to understand what this error is for

Reply:
Sorry I just saw the comment. I will address that in the other branch @aattuluri